literal/decimal_util_test.go

```diff
@@ -0,0 +1,103 @@
+package literal
+
+import (
+	"fmt"
+	"math/big"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_decimalStringToBytes(t *testing.T) {
+	tests := []struct {
+		input        string
+		hexWant      string
+		expPrecision int32
+		expScale     int32
+	}{
+		{"12345", "39300000000000000000000000000000", 5, 0},
+		{"+12345", "39300000000000000000000000000000", 5, 0},
+		{"-12345", "c7cfffffffffffffffffffffffffffff", 5, 0},
+		{"123.45", "39300000000000000000000000000000", 5, 2},
+		{"-123.45", "c7cfffffffffffffffffffffffffffff", 5, 2},
+		{"0.123", "7b000000000000000000000000000000", 4, 3},
+		{"-0.123", "85ffffffffffffffffffffffffffffff", 4, 3},
+		{"9223372036854775807", "ffffffffffffff7f0000000000000000", 19, 0},  // Max int64
+		{"-9223372036854775808", "0000000000000080ffffffffffffffff", 19, 0}, // Min int64
+		{"99999999999999999999999999999999999999", "ffffffff3f228a097ac4865aa84c3b4b", 38, 0},
+		{"+99999999999999999999999999999999999999", "ffffffff3f228a097ac4865aa84c3b4b", 38, 0},
+		{"-99999999999999999999999999999999999999", "01000000c0dd75f6853b79a557b3c4b4", 38, 0},
+		{"0", "00000000000000000000000000000000", 1, 0},
+		{"-0", "00000000000000000000000000000000", 1, 0},
+		{"0.0", "00000000000000000000000000000000", 2, 1},
+		{"65535", "ffff0000000000000000000000000000", 5, 0},
+		{"-65535", "0100ffffffffffffffffffffffffffff", 5, 0},
+		{"18446744073709551615", "ffffffffffffffff0000000000000000", 20, 0},  // Max uint64
+		{"-18446744073709551616", "0000000000000000ffffffffffffffff", 20, 0}, // Min int64 - 1
+		{"12345.6789", "15cd5b07000000000000000000000000", 9, 4},
+		{"1234567890123456", "c0ba8a3cd56204000000000000000000", 16, 0},
+		{"1234567890123456.78901234", "f2af966ca0101f9b241a000000000000", 24, 8},
+		{"0.0012345678901234", "f22fce733a0b00000000000000000000", 17, 16},
+		{"-0.0012345678901234", "0ed0318cc5f4ffffffffffffffffffff", 17, 16},
+		{"123456789012345678901234567890.1234", "f2af967ed05c82de3297ff6fde3c0000", 34, 4},
+		{"-1234567890.1234567890", "2ef5e0147356ab54ffffffffffffffff", 20, 10},
+		{"0", "00000000000000000000000000000000", 1, 0},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.input, func(t *testing.T) {
+			testDecimalStringToBytes(t, tt.input, tt.hexWant, tt.expPrecision, tt.expScale)
+		})
+	}
+
+	badInputs := []string{
+		"12345678901234567890123456789012345678901234",
+		"abc",
+		"12.34.56",
+		"199999999999999999999999999999999999999",
+		"1.23e20",
```

Comment:
This should be a valid input IMHO, why would this be invalid?

Reply:
Not sure about this. ChatGpt says it is not a valid literal for decimal type.
I tried this on duckdb: select `1023e23` it recognised this as `double`. When I tried to insert this in a `decimal(38, 9)` column, it actually stored `123000000000000003008.758808576`
Postgres support this notation.
apd.NewFromString also supports this notation.

Reply:
I think it is okay for our utility library to provide more options for input than what a random database may provide. They are utility functions afterall. We should just document what is allowed.

Reply:
added